src/libstd/sys/unix/os.rs

```diff
@@ -423,12 +422,10 @@ pub fn env() -> Env {
             }
             environ = environ.offset(1);
         }
-        let ret = Env {
+        return Env {
```

Comment:
This `return` is unnecessary, right?

Reply:
Unfortunately the inner function definition `fn parse()` below confuses the rustc parser if we omit the `return`

Reply:
Oh, I see. This is pretty weird - I'd have expected the inner function to precede the actual code of the function (or to be inlined, since it has just one caller).

Thanks for explaining.